kinbot/pes.py

```diff
@@ -931,84 +1010,187 @@ def create_mess_input(par, wells, products, reactions, barrierless,
+                if i == 0:
+                    freq_0 = make_freq(prod, frequencies, freq_factor, parent[prod], 0, par['high_level'])
+                    nfreq_0 = len(frequencies)
+                elif i == 1:
+                    freq_1 = make_freq(prod, frequencies, freq_factor, parent[prod], 0, par['high_level'])
+                    nfreq_1 = len(frequencies)
+
+                # check for rotor uq file
+                name = os.getcwd() + '/' + parent[prod] + '/uq_' + p + "_rotor.txt"
+                dest = os.getcwd() + '/uq_' + p + "_rotor.txt"
+                try:
+                    shutil.copyfile(name, dest)
+                except IOError:
+                    pass
+
             name = pr_short[prod] + ' ! ' + prod
-            energy = prod_energies[prod]
+            if par['L3_calc'] == 1:
+                energy = prod_l3energies[prod]
+            else: 
+                energy = prod_energies[prod]
+            uq_energyAdd, energy_nromfactor = uq_obj.calc_factor('energy', pr_short[prod], uq_iter, 1)
+            uq_obj.write_uqtk_data("energy", energy_normfactor, prod, uq_iter)
+            energy = energy + uq_energyAdd
+            all_energies[prod] = energy
+
             fr_names = {}
             for fr in prod.split('_'):
                 key = 'fr_name_{}'.format(fr)
                 value = fr_short[fr] + ' ! ' + fr
                 fr_names[key] = value
-            with open(parent[prod] + '/' + prod + '_0000.mess') as f:
-                s.append(f.read().format(name=name,
+            reaction_items.append(prod)
+            mess_iter = "{0:04d}".format(uq_iter)
+            # check to make sure values match
+            for i, p in enumerate(prods):
+                atoms = sq.get_sql_mess_data(all_data, p, level, 'atoms')
+                natoms = len(atoms)
+                frequencies = sq.get_sql_mess_data(all_data, p, level, 'red_freq')
+                nfreq = len(frequencies)
+                # DEBUGGING
+                n = p + "_values.txt"
+                f = open(n, 'w')
+                f.write("p: {}\nfreqs: {}\nnfreq: {}".format(p, frequencies, nfreq))
+                f.write("i: {}\nfreq0: {}\nnfreq0: {}\nfreq1: {}\nnfreq1: {}".format(i, freq_0, nfreq_0, freq_1, nfreq_1))
+                if i == 0:
+                    if natoms == 1:
+                        nfreq_tmp = nfreq_0
+                        nfreq_0 = nfreq_1
+                        nfreq_1 = nfreq_tmp
+                        freq_tmp = freq_0
+                        freq_0 = freq_1
+                        freq_1 = freq_tmp
+                elif i == 1:
+                    if natoms == 1: 
+                        nfreq_tmp = nfreq_1
+                        nfreq_1 = nfreq_0
+                        nfreq_0 = nfreq_tmp
+                        freq_tmp = freq_1
+                        freq_1 = freq_0
+                        freq_0 = freq_tmp
+                f.close() 
```

Comment:
lines 1150-1166 ensure that the fragments are printed properly into the MESS file in the case where a single atom is part of the bimolecular prod.